Authenticator logic

mackozer opened the pull request:
Authentication with OAuthSwift is done.
Added:
+ Dependencies class (initialized in AppFlowCoordinator)
+ UserManager class (initialised in Dependencies)

mackozer committed:
OAuthSwift added

OauthSwift Pod added. Authorization class and Authorization strings added. AppDelegate aplication(open:url) added.

mackozer committed:
OAuth authorization.

OAuth authrozation with API works.

mackozer committed:
OAuth login works.

Getting user access token.

mackozer committed:
Authorization is done

Authorization is done.

Siemian commented on Project/Source Files/Services/Networking/UserManager.swift:
```diff
@@ -0,0 +1,11 @@
+
+import Foundation
+
+class UserManager {
```
final missing

Siemian commented on Project/Source Files/Services/Networking/Authorization.swift:
```diff
@@ -0,0 +1,33 @@
+import Foundation
+import OAuthSwift
+
+class Authorization {
```
final missing

Siemian commented on Project/Application/Dependencies.swift:
```diff
@@ -0,0 +1,11 @@
+
+import Foundation
+
+class Dependencies {
```
final missing

Siemian commented on Project/Application/Dependencies.swift:
```diff
@@ -0,0 +1,11 @@
+//  Dependencies.swift
+//  NetguruCollegeApp
+//
+
```
redundant new line, one is enough
please fix here in all files

Siemian commented on Project/Source Files/Services/Networking/Authorization.swift:
```diff
@@ -0,0 +1,33 @@
+            withCallbackURL: URL(string: AuthorizationStrings.callback)!, scope: "repo+user", state: "Github") { result in
+                switch result {
+                case .success(let (credential, _, _)):
+                    print("Oauth token: \(credential.oauthToken)")
```
prints should not be pushed to the repo

Siemian commented on Project/Source Files/Services/Networking/Authorization.swift:
```diff
@@ -0,0 +1,33 @@
+                    completion(credential.oauthToken, nil)
+                // Do your request
+                case .failure(let error):
+                    print(error.localizedDescription)
```
same here

mackozer committed:
KeychainSwift pod added

KeychainSwift pod added

mackozer committed:
Saving token to keychain, code cleanup

Auth token saved or read to/from keychain.
Code cleanup (added final to classes, deleted print calls).

Siemian commented on Project/Source Files/Services/Networking/Authorization.swift:
```diff
@@ -0,0 +1,30 @@
+            consumerSecret: AuthorizationStrings.clientSecret,
+            authorizeUrl: AuthorizationStrings.authorizeURL,
+            accessTokenUrl: AuthorizationStrings.accessTokenURL,
+            responseType: AuthorizationStrings.token)
```
please move `)` to new line

mackozer committed:
Merge branch 'develop' into authenticator_logic

mackozer committed:
code cleanup

code cleanup

mackozer committed:
Merge branch 'authenticator_logic' of https://github.com/netguru-college/ios-lodz19-control into authenticator_logic
